### DISCUSSION TITLE
altered pdf file is still validated

### DISCUSSION THREAD

**Discussion by rachmanmarjianto** [lsnepomuceno/laravel-a1-pdf-sign#140] [Q&A]
I tried to validate an altered PDF and the validation's result is still true. Then I checked your validation code. 
This line gives the validation result --> isValidated : true
$finalContent['validated'] = !!count(array_intersect_key(array_flip(['OU', 'CN']), $finalContent));

I don't understand, what does the variable isValidated in ValidatedSignedPDF's object mean?

**Comment by lsnepomuceno** [comment]
Hello @rachmanmarjianto,

You read the code correctly, and the conclusion is the one you suspected: **`isValidated` did not mean the signature was valid.** That line checked whether the parsed certificate subject happened to contain a `CN` or `OU` field — which a tampered document still has, because altering the page bytes does not remove the signer's name from the embedded certificate.

So the field answered "is there a certificate in here that looks like a certificate", not "does this signature still match these bytes". Your altered PDF was reported as validated for exactly that reason. This was a real defect, not a misunderstanding on your side, and thank you for taking the trouble to trace it to the line.

**Fixed in 2.0.** Validation now verifies the CMS cryptographically against the byte range each signature covers:

```php
use LSNepomuceno\LaravelA1PdfSign\Facades\A1PdfSign;

$report = A1PdfSign::validate('path/to/document.pdf');

$report->isValid();    // every signature verifies against the bytes it covers
$report->count();      // all signatures, not only the first
$report->signers();    // the signer identity as structured data
$report->latest();
```

An altered document now returns `false`. There is a test that does precisely what you did — signs a document, changes a byte, and asserts the report turns invalid.

Two things worth knowing about the new meaning:

- `isValid()` answers "does this signature match these bytes". It deliberately does **not** check the issuer against a trust store — whether you trust the CA is your application's policy, not the package's.
- Documents carrying several signatures are reported in full. 1.x read only the first.

2.0 requires PHP 8.4 and Laravel 13, and the API changed — [UPGRADE.md](https://github.com/lsnepomuceno/laravel-a1-pdf-sign/blob/main/UPGRADE.md) maps every old call to its replacement.

Sorry this took so long to answer.
